File: 1_TwoSum.js

let twoSum = function(arr,target) {

    let result = [];
    for(let i = 0; i < arr.length; i++) {

        for(let j = i+1; j < arr.length; j++) {
            
            if(arr[i] + arr[j] == target) {
                result.push(i);
                result.push(j);
                return result;
            }
        }
    }
    return null;
}

var twoSum1 = function(nums, target) {

    let map = {}
    for(let i = 0 ; i < nums.length ; i++){
        let diff = target-nums[i]
        if(diff in map){
            return [map[diff] , i]
        }
        map[nums[i]] = i 
    }
};

let main = function() {

    let arr = [1,2,3,4,5];
    let target = 8;

    let result = twoSum1(arr,target)
    console.log(result);
}

main();

File: 1_twoSum.py

class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        seen = {} # number -> idx

        for i in range(len(nums)):
            diff = target - nums[i]
            if diff in seen:
                return [i, seen[diff]]
            seen[nums[i]] = i

        return [-1,-1]


nums = [2,7,11,15]
target = 9

obj = Solution()
print(obj.twoSum(nums,target))